specs/lib/removeUnusedVerticesSpec.js

```diff
@@ -0,0 +1,254 @@
+'use strict';
+var clone = require('clone');
+var bufferEqual = require('buffer-equal');
+var removeUnusedVertices = require('../../lib/removeUnusedVertices');
+
+describe('removeUnusedVertices', function() {
+    var testBuffer = new Buffer(840);
+    var gltf = {
+        "buffers": {
+            "buffer_0": {
+                "uri": "data:,",
+                "byteLength": 840,
+                "extras": {
+                    "_pipeline": {
+                        "source": testBuffer
+                    }
+                }
+            }
+        }
+    };
+
+    it('does not remove any data with a single buffer view', function() {
+        var testGltf = clone(gltf);
+        testGltf.bufferViews = {
+            "bufferView_0": {
+                "buffer": "buffer_0",
+                "byteOffset": 0,
+                "byteLength": 840
+            }
+        };
+        
+        removeUnusedVertices(testGltf);
+
+        delete testGltf.buffers.buffer_0.extras._pipeline.offset;
+        delete testGltf.buffers.buffer_0.extras._pipeline.end;
+        expect(bufferEqual(testGltf.buffers.buffer_0.extras._pipeline.source, testBuffer)).toBe(true);
+    });
+
+    it('does not remove any data with overlapping buffer views', function() {
+        var testGltf = clone(gltf);
+        testGltf.bufferViews = {
+            "bufferView_0": {
+                "buffer": "buffer_0",
+                "byteOffset": 0,
+                "byteLength": 360
+            },
+            "bufferView_1": {
+                "buffer": "buffer_0",
+                "byteOffset": 240,
+                "byteLength": 600
+            }
+        };
+
+        removeUnusedVertices(testGltf);
+
+        delete testGltf.buffers.buffer_0.extras._pipeline.offset;

```

Comment:
Actually, turns out I don't need these lines as I was using them in a test which I deleted.